generated/nidaqmx/nidaqmx_service.cpp

```diff
@@ -7272,17 +7287,30 @@ namespace nidaqmx_grpc {
       return ::grpc::Status::CANCELLED;
     }
     try {
-      uInt32 buffer_size = request->buffer_size();
-      std::string error_string;
-      if (buffer_size > 0) {
-          error_string.resize(buffer_size-1);
-      }
-      auto status = library_->GetExtendedErrorInfo((char*)error_string.data(), buffer_size);
-      response->set_status(status);
-      if (status == 0) {
-        response->set_error_string(error_string);
+
+      while (true) {
+        auto status = library_->GetExtendedErrorInfo(nullptr, 0);
+        if (status < 0) {
+          response->set_status(status);
+          return ::grpc::Status::OK;
+        }
+        uInt32 buffer_size = status;
+      
+        std::string error_string;
+        if (buffer_size > 0) {
+            error_string.resize(buffer_size-1);
+        }
+        status = library_->GetExtendedErrorInfo((char*)error_string.data(), buffer_size);
+        if (status == kErrorReadBufferTooSmall || status > buffer_size) {
```

Comment:
In DAQ there's also a `DAQmxErrorBufferTooSmallForString`. Is it possible that could be returned here?

Or do these actually have the behavior where they don't return an error?

We can test these specific methods with the old implementation by passing in a too-small-non-zero size.

Reply:
I tested it it out and it actually returns a warning: `kWarningCAPIStringTruncatedToFitBuffer`. This is used in some common places where we're returning a string from an API, so I think it makes sense to handle it explicitly. (note that the code as is will probably handle this correctly since it's a positive fairly large value, but I guess it would fail for a > 200000 character string...)